lib/xdr/accounts/thresholds.ex

```diff
@@ -0,0 +1,79 @@
+defmodule Stellar.XDR.Thresholds do
+  @moduledoc """
+  Representation of an AccountEntry's Thresholds.
+  Thresholds stores unsigned bytes: [weight of master|low|med|high]
+  """
+  alias Stellar.XDR.Opaque4
+
+  @behaviour XDR.Declaration
+
+  @type thresholds :: [
+          master_weight: non_neg_integer(),
+          low: non_neg_integer(),
+          med: non_neg_integer(),
+          high: non_neg_integer()
+        ]
+
+  @type t :: %__MODULE__{thresholds: thresholds}
```

Comment:
Using a keyword list looks like over-engineering the implementation. We can work only with the Struct:

```elixir
    @type t :: %__MODULE__{
          master_weight: non_neg_integer(),
          low: non_neg_integer(),
          med: non_neg_integer(),
          high: non_neg_integer()}
```

Reply:
I thought it would be a good idea to use the keyword list to force the user to set values for every threshold. 
What if we use a simple parameter list instead, specifying each threshold datatype. Something like:

``` elixir
@spec new(master_weight :: byte(), low :: byte(), med :: byte(), high :: byte()) :: t()
def new (master_weight, low, med, high)...
```

And then we proceed pattern match according to the struct.

Reply:
@juanhurtado10 you have more experience building the library; what do you think?

Reply:
I will come up with a simpler solution for the next commit!

Reply:
The Stellar.XDR.Thresholds module's struct was flattened to get the rid of the complexity added by a keyword list. The module's constructor keeps the keyword list for the parameters but an error is now raised to provide a hint to API users on how to use the constructor.

Reply:
@luishurtado @JuanCamiloPerilla

1. Definitely agree with Luis, let's use the Struct arguments.

2. In my opinion, using a keyword list makes the code more declarative than just having integer arguments. This is probably easier to read and understand:
```elixir
Stellar.XDR.Tresholds.new(master_weight: 1, low: 1, med: 2, high: 3)...
```

than this

```elixir
Stellar.XDR.Tresholds.new(1, 1, 2, 3)...
```

Best!

Reply:
Changes applied!